src/azure-cli/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -166,6 +166,27 @@ def validate_policy_permissions(ns):
             '--certificate-permissions --storage-permissions')
 
 
+def validate_private_endpoint_connection_id(cmd, ns):
+    connection_id = ns.connection_id
+    connection_name = ns.private_endpoint_connection_name
+    vault_name = ns.vault_name
+
+    if not connection_id:
+        if not all([connection_name, vault_name]):
+            raise argparse.ArgumentError(
```

Comment:
Based on my knowledge, usually we would `raise CLIError('usage error:')`. We can discuss.